src/components/vm/nics/vmNicsCard.tsx

```diff
@@ -269,9 +282,130 @@ interface IpInterface {
+    id: string,
+    networkId: number,
+}) => {
+    const inetIps: IpAddress[] = [];
+    const inet6Ips: IpAddress[] = [];
+
+    ipInterfaces.forEach(ipInterface => {
+        ipInterface.ips.forEach(inet => {
+            // Separate out IPv4 and IPv6 and don't add duplicate IPs.
+            if (inet.type === IpAddressVersion.v4 && !inetIps.some(inetIp => inetIp.ip == inet.ip)) {
+                inetIps.push(inet);
+            } else if (inet.type === IpAddressVersion.v6 && !inet6Ips.some(inet6Ip => inet6Ip.ip == inet.ip)) {
+                inet6Ips.push(inet);
+            }
+        });
+    });
+
+    if (inetIps.length === 0 && inet6Ips.length === 0) {
+        // There is not IP address associated with this NIC
+        return null;
+    }
+
+    return (
+        <>
+            {inetIps.length > 0 &&
+                <DescriptionListGroup>
+                    {inetIps.map((ip, index) => (
+                        <React.Fragment key={ip.ip}>
+                            <DescriptionListTerm>
+                                {_("IPv4 Address")}
+                            </DescriptionListTerm>
+                            <DescriptionListDescription id={`${id}-network-${networkId}-ipv4-address-${index}`}>
+                                {ip.ip}/{ip.prefix}
+                            </DescriptionListDescription>
+                        </React.Fragment>
+                    ))}
+                </DescriptionListGroup>
+            }
+            {inet6Ips.length > 0 &&
+                <DescriptionListGroup>
+                    {inet6Ips.map((ip, index) => (
+                        <React.Fragment key={ip.ip}>
+                            <DescriptionListTerm>
+                                {_("IPv6 Address")}
+                            </DescriptionListTerm>
+                            <DescriptionListDescription id={`${id}-network-${networkId}-ipv6-address-${index}`}>
+                                {ip.ip}/{ip.prefix}
+                            </DescriptionListDescription>
+                        </React.Fragment>
+                    ))}
+                </DescriptionListGroup>
+            }
+        </>
+    );
+};
+
+const IPTarget = ({
+    ipInterfaces,
+    id,
+} : {
+    ipInterfaces: IpInterface[],
+    id: string,
+}) => {
+    for (const iface of ipInterfaces) {
+        if (iface.source == "agent")
+            return <div id={id}>{iface.name}</div>;
+    }
+
+    return null;
+};
+
+const IPAbbrev = ({
+    ipInterfaces,
+} : {
+    ipInterfaces: IpInterface[],
+}) => {
+    /* Return the first IPv4 address, or the first IPv6 address.
+     */
+
+    for (const iface of ipInterfaces) {
+        for (const ip of iface.ips) {
+            if (ip.type == IpAddressVersion.v4)
+                return ip.ip;
+        }
+    }
+
+    for (const iface of ipInterfaces) {
+        for (const ip of iface.ips) {
+            if (ip.type == IpAddressVersion.v6)
+                return ip.ip;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2324-02ce13bf-20250922-141831-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)